node/bft/src/bft.rs

```diff
@@ -809,7 +809,7 @@ impl<N: Network> BFT<N> {
             // Acquire the BFT write lock.
             let mut dag = self.dag.write();
             // Iterate over the certificates.
-            for certificate in certificates {
+            for certificate in certificates.clone() {
```

Comment:
we should probably iterate by reference only clone in case of insertion

Reply:
resolved in 073dab41f